### PR TITLE
Fix issue 458: Add null checking

### PR DESCRIPTION
**Pull request by arthurscchan**
This PR fixes issue #458 by adding a null check before accessing the length of the text string. This could avoid errors when getDecimalValue is invoked when there is no more data left in the input.

### CODE REVIEW

File: cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORParser.java [hunk 1]
```diff
@@ -2224,6 +2224,9 @@ protected void convertNumberToBigDecimal() throws IOException
             // Let's parse from String representation, to avoid rounding errors that
             //non-decimal floating operations would incur
             final String text = getText();
+            if (text == null) {
+                throw _constructReadException("No more values");
```
Comment (cowtowncoder) [inline cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORParser.java:2228]: Ok; I'll see what exactly is going on here.

**Review by cowtowncoder — APPROVED** [review]
With minor change to fix, LGTM, approved, will merge.

File: cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORParser.java [hunk 2]
```diff
@@ -1112,6 +1112,9 @@ protected JsonToken _handleTaggedBinary(TagList tags) throws IOException
         } else {
             // 12-May-2016, tatu: Since that's all we know, let's otherwise
             //   just return default Binary data marker
+            // 16-Jan-2024, tatu: Esoteric edge case where we have marked
+            //   `int` as being tokenized
+            _numTypesValid = NR_UNKNOWN;
```
Comment (cowtowncoder) [inline cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORParser.java:1117]: This is the actual fix: for some reason `int` tag indicated number token. May need to revisit logic at some point but this should work for now.

**Comment by arthurscchan** [comment]
Thanks for showing me the details of the problematic part.

**Pull request merged**